pdns/pdnsutil.cc

```diff
@@ -68,6 +68,519 @@ namespace {
+     "ZONE ALGORITHM {ksk|zsk} MODULE SLOT PIN LABEL [PUBLABEL]",
+     "\tAssign a Hardware Signing Module to a ZONE"}},
+   {"create-key", {true, HSMCreateKey,
+     "ZONE KEY_ID [BITS]",
+     "\tcreate a key using Hardware Signing Module for ZONE (use assign first);\n"
+     "\tBITS defaults to 2048"}}}
+};
+#endif // ]
+
+// META/МETADATA
+
+static const groupCommandDispatcher metadataCommands{
+  "Zone Metadata",
+  {{"add", {true, addMeta,
+    "ZONE KIND VALUE [VALUE...]",
+    "\tAdd zone metadata, this adds to the existing KIND"}},
+   {"get", {true, getMeta,
+    "ZONE [KIND...]",
+    "\tGet zone metadata. If no KIND is given, lists all known"}},
+   {"set", {true, setMeta,
+    "ZONE KIND [VALUE...]",
+    "\tSet zone metadata, replacing all existing records of KIND, optionally\n"
+    "\tproviding a value. An omitted value clears the metadata"}}}
+};
+
+// NETWORKS (VIEWS CONTEXT)
+
+static const groupCommandDispatcher networkCommands{
+  "Networks",
+  {{"list", {true, listNetwork,
+    "",
+    "\tList all defined networks with their chosen views"}},
+   {"set", {true, setNetwork,
+    "NET [VIEW]",
+    "\tSet the view for a network, or delete if no view argument."}}}
+};
+
+// RECORD/RRSET
+
+static const groupCommandDispatcher rrsetCommands{
+  "Zone Record",
+  {{"add", {true, addRecord,
+    R"(ZONE NAME TYPE [TTL] "CONTENT" ["CONTENT"...])",
+    "\tAdd one or more records to the given rrset in ZONE"}},
+   {"delete", {true, deleteRRSet,
+    "ZONE NAME TYPE",
+    "\tDelete named rrset from ZONE"}},
+   {"hash", {true, hashZoneRecord,
+    "ZONE NAME",
+    "\tCalculate the NSEC3 hash for NAME in ZONE"}},
+   {"replace", {true, replaceRRSet,
+    R"(ZONE NAME TYPE [TTL] "CONTENT" ["CONTENT"...])",
+    "\tReplace named rrset from ZONE"}}}
+};
+
+// TSIG-KEY / TSIGKEY
+
+static const groupCommandDispatcher TSIGKEYCommands{
+  "TSIG Key",
+  {{"activate", {true, activateTSIGKey,
+    "ZONE NAME {primary|secondary|producer|consumer}",
+    "\tEnable TSIG authenticated AXFR using the key NAME for ZONE"}},
+   {"deactivate", {true, deactivateTSIGKey,
+    "ZONE NAME {primary|secondary|producer|consumer}",
+    "\tDisable TSIG authenticated AXFR using the key NAME for ZONE"}},
+   {"delete", {true, deleteTSIGKey,
+    "NAME",
+    "\tDelete TSIG key (warning: will not unmap key!)"}},
+   {"generate", {true, generateTSIGKey,
+    "NAME ALGORITHM",
+    "\tGenerate new TSIG key.\n"
+    "\tALGORITHM is one of hmac-{md5,sha1,sha224,sha256,sha384,sha512}"}},
+   {"import", {true, importTSIGKey,
+    "NAME ALGORITHM KEY",
+    "\tImport TSIG key"}},
+   {"list", {true, listTSIGKeys,
+    "",
+    "\tList all TSIG keys"}}}
+};
+
+// VIEWS
+
+static const groupCommandDispatcher viewsCommands{
+  "Views",
+  {{"list", {true, listView,
+    "",
+    "\tList all zones within VIEW"}},
+   {"list-all", {true, listViews,
+    "",
+    "\tList all view names"}},
+   {"add-zone", {true, viewAddZone,
+    "VIEW ZONE..VARIANT",
```

Comment:
I caught myself thinking "this should be `ZONE[..VARIANT]`" but the unvarianted zone will appear anyway. Now I wonder if this poses a challenge/risk of confusion for users, especially with frontends like ZC. Not a question for this PR really.